src/libraries/System.Private.CoreLib/src/System/Random.Xoshiro256StarStarImpl.cs

```diff
@@ -90,46 +95,46 @@ public override int Next()
 
             public override int Next(int maxValue)
             {
-                if (maxValue > 1)
+                Debug.Assert(maxValue >= 0);
+
+                ulong randomProduct = (ulong)(uint)maxValue * NextUInt32();
+                uint lowPart = (uint)randomProduct;
+
+                if (lowPart < (uint)maxValue)
                 {
-                    // Narrow down to the smallest range [0, 2^bits] that contains maxValue.
-                    // Then repeatedly generate a value in that outer range until we get one within the inner range.
-                    int bits = BitOperations.Log2Ceiling((uint)maxValue);
-                    while (true)
+                    uint remainder = ((uint)0 - (uint)maxValue) % (uint)maxValue;
```

Comment:
`(uint)0` can be `0u`, `(ulong)0` can be `0ul`, etc.

Reply:
Fixed.